src/Middleware/RequestDecompression/src/DecompressionProviderCollection.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
We borrowed this pattern from MVC. Something about the generic syntax and adding implementations by type instead of by instance. That allows them to be DI activated.